src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -440,11 +413,64 @@ private static string RemoveTrailingZerosFromMillisecondsForAGivenDate(DateTimeO
             return formattedDate.Replace(milliseconds, trimmedMilliseconds, StringComparison.Ordinal);
         }
 
-        private static string RemoveVersionIdAndLastUpdatedFromMeta(ResourceWrapper resourceWrapper)
+        private void ReplaceVersionIdAndLastUpdatedInMeta(ResourceWrapper resourceWrapper)
+        {
+            var date = GetJsonValue(resourceWrapper.RawResource.Data, "lastUpdated");
+            string rawResourceData;
+            if (resourceWrapper.Version == "1") // version is already correct
+            {
+                rawResourceData = resourceWrapper.RawResource.Data
+                                    .Replace($"\"lastUpdated\":\"{date}\"", $"\"lastUpdated\":\"{RemoveTrailingZerosFromMillisecondsForAGivenDate(resourceWrapper.LastModified)}\"", StringComparison.Ordinal);
+            }
+            else
+            {
+                var version = GetJsonValue(resourceWrapper.RawResource.Data, "versionId");
+                rawResourceData = resourceWrapper.RawResource.Data
+                                    .Replace($"\"versionId\":\"{version}\"", $"\"versionId\":\"{resourceWrapper.Version}\"", StringComparison.Ordinal)
+                                    .Replace($"\"lastUpdated\":\"{date}\"", $"\"lastUpdated\":\"{RemoveTrailingZerosFromMillisecondsForAGivenDate(resourceWrapper.LastModified)}\"", StringComparison.Ordinal);
+            }
+
+            resourceWrapper.RawResource = new RawResource(rawResourceData, FhirResourceFormat.Json, true);
+        }
+
+        private bool ExistingRawResourceIsEqualToInput(ResourceWrapper input, ResourceWrapper existing) // call is not symmetrical, it assumes version = 1 on input.
+        {
+            var inputDate = GetJsonValue(input.RawResource.Data, "lastUpdated");
+            var existingDate = GetJsonValue(existing.RawResource.Data, "lastUpdated");
+            var existingVersion = GetJsonValue(existing.RawResource.Data, "versionId");
+            if (existingVersion != "1")
+            {
+                return input.RawResource.Data == existing.RawResource.Data.Replace($"\"lastUpdated\":\"{existingDate}\"", $"\"lastUpdated\":\"{inputDate}\"", StringComparison.Ordinal);
+            }
+            else
+            {
+                return input.RawResource.Data
+                            == existing.RawResource.Data
+                                    .Replace($"\"versionId\":\"{existingVersion}\"", $"\"versionId\":\"1\"", StringComparison.Ordinal)
+                                    .Replace($"\"lastUpdated\":\"{existingDate}\"", $"\"lastUpdated\":\"{inputDate}\"", StringComparison.Ordinal);
+            }
+        }
+
+        private string GetJsonValue(string json, string propName)
```

Comment:
I've noticed this method only works with properties with values between quotes (and no spaces between the ":" char). 
Is that expected?

Per example, the following statement returns an empty string:
string value = GetJsonValue("{ \"version\" : \"1\" }", "version");

Reply:
True. This works for our current raw resource formatting only which does not have extra spaces.
All this logic should be thrown away once we have good solution for this bug [https://microsofthealth.visualstudio.com/Health/_workitems/edit/99962/](https://microsofthealth.visualstudio.com/Health/_workitems/edit/99962/)